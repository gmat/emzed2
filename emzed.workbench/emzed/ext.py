import pkg_resources

for ep in pkg_resources.iter_entry_points("emzed_plugin", name="package"):
    print ep
    package = ep.load()
    exec "%s=package" % package.__name__
    del package
    del ep
del pkg_resources

