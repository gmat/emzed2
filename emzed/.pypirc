[distutils]
index-servers = 
  local

[local]
repository: http://127.0.0.1:8080
username:uschmitt
password:pillepalle

