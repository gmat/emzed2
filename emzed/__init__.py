# encoding:latin-1

from version import version as __version__


