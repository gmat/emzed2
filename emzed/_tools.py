
def is_started_from_cmdline():

    import inspect
    info = inspect.stack()[-1][1]
    result = False
    if info != "<string>":
        if info.endswith("emzed.workbench"):
            result = True
    return result


def gui_running():
    from PyQt4.QtGui import QApplication
    return QApplication.instance() is not None


def is_first_start():
    from emzed.core.config import _UserConfig
    import os
    return not os.path.exists(_UserConfig.config_file_path())
