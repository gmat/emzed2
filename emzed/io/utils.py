

def _prepare_path(path, extensions):

    import sys
    if isinstance(path, unicode):
        path = path.encode(sys.getfilesystemencoding())
    elif path is None:
        from .. import gui
        path = gui.askForSingleFile(extensions=extensions)
    return path
