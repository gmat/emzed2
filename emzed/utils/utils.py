import emzed.core.data_types

def toTable(colName, iterable,  format_="", type_=None, title="", meta=None):
    return emzed.core.data_types.Table.toTable(colName, iterable, format_, type_, title, meta)

toTable.__doc__ = emzed.core.data_types.Table.toTable.__doc__

def formula(mf):
    """
    Creates formula object which allows addition and subtraction:

    .. pycon::

       import ms   !onlyoutput
       mf1 = ms.formula("H2O")
       mf2 = ms.formula("NaOH")
       mf3 = mf1 + mf2
       print str(mf3)
       print str(mf3 - mf1)

    Mass calculation is supported too:

    .. pycon::

       print mf1.mass()

    If you need some internal representation, you can get a dictionary.
    Keys are pairs of *(symbol, massnumber)*, where *massnumber = None*
    refers to the lowest massnumber. Values of the dictionary are counts:

    .. pycon::

       print mf1.asDict()
       mixed = ms.formula("[13]C2[14]C3")
       print mixed.asDict()


    """
    from emzed.core.chemistry import MolecularFormula
    return MolecularFormula(mf)

def addmf(formula0, *formulas):
    """
    Combines molecular formulas by addition and subtraction:

    .. pycon::

       import ms !onlyoutput
       print ms.addmf("H2O", "COOH")
       print ms.addmf("H2O", "COOH", "NaCl")

    A leading minus sign subtracts the formula following this sign:

    .. pycon::

       print ms.addmf("H2O2", "-H2O")
       print ms.addmf("H2O", "COOH", "NaCl", "-H2O2")
       print ms.addmf("(CH2)7COOH", "-C7")

    """

    mf0 = formula(formula0)
    for f in formulas:
        if f.startswith("-"):
            mf0 -= formula(f[1:])
        else:
            mf0 += formula(f)
    return str(mf0)


###


def openInBrowser(urlPath):
    """
    opens *urlPath* in browser, eg:

    .. pycon::
        ms.openInBrowser("http://emzed.biol.ethz.ch") !noexec

    """
    from PyQt4.QtGui import QDesktopServices
    from PyQt4.QtCore import QUrl
    import os.path

    url = QUrl(urlPath)
    scheme = url.scheme()
    if scheme not in ["http", "ftp", "mailto"]:
        # C:/ or something simiar:
        if os.path.splitdrive(urlPath)[0] != "":
            url = QUrl("file:///"+urlPath)
    ok = QDesktopServices.openUrl(url)
    if not ok:
        raise Exception("could not open '%s'" % url.toString())


def _recalculateMzPeakFor(postfix):
    def calculator(table, row, name, postfix=postfix):

        mzmin = table.get(row, "mzmin"+postfix)
        mzmax = table.get(row, "mzmax"+postfix)
        rtmin = table.get(row, "rtmin"+postfix)
        rtmax = table.get(row, "rtmax"+postfix)
        pm    = table.get(row, "peakmap"+postfix)
        mz = pm.representingMzPeak(mzmin, mzmax, rtmin, rtmax)
        return mz if mz is not None else (mzmin+mzmax)/2.0
    return calculator

def _hasRangeColumns(table, postfix):
    return all([table.hasColumn(n + postfix) for n in ["rtmin", "rtmax",
                                                 "mzmin", "mzmax", "peakmap"]])

def recalculateMzPeaks(table):
    #TODO: tests !
    """Adds mz value for peaks not detected with centwaves algorithm based on
       rt and mz window: needed are columns mzmin, mzmax, rtmin, rtmax and
       peakmap mz, postfixes are automaticaly taken into account"""
    postfixes = [ "" ] + [ "__%d" % i for i in range(len(table._colNames))]
    for postfix in postfixes:
        if _hasRangeColumns(table, postfix):
            mz_col = "mz" + postfix
            if table.hasColumn(mz_col):
                table.replaceColumn(mz_col, _recalculateMzPeakFor(postfix),
                                    format="%.5f", type_=float)
            else:
                table.addColumn(mz_col, _recalculateMzPeakFor(postfix),
                                format="%.5f", type_=float)

def startfile(path):
    import sys, os
    if sys.platform == "win32":
        path = path.replace("/", "\\") # needed for network paths
    return os.startfile(path)

