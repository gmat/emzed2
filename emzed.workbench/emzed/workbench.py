def run():
    print 42

def main():
    import os
    print os.system("pip install -i http://127.0.0.1:3142/root/dev/+simple emzed.core")
    import emzed.core
    print emzed.core



