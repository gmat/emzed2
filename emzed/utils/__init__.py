from utils import *
from isotope_calculator import isotopeDistributionTable, plotIsotopeDistribution
from formula_generator  import formulaTable

from load_utils import loadCSV, loadTable, loadPeakMap

from feature_detectors import runCentwave, runMatchedFilters, runMetaboFeatureFinder

from metlin import matchMetlin

from integration import integrate
