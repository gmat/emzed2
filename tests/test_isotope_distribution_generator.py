import numpy

import emzed.utils


def testIDGen(regtest):
    t = emzed.utils.isotopeDistributionTable("S4C4", R=50000)
    t._print(out=regtest)

    t = emzed.utils.isotopeDistributionTable("S4C4", R=10000)

    t._print(out=regtest)

    t = emzed.utils.isotopeDistributionTable("S4C4", R=10000, fullC13=True)
    t._print(out=regtest)

    t = emzed.utils.isotopeDistributionTable("C4", R=10000, C=dict(C12=0.5, C13=0.5))
    t._print(out=regtest)

    t = emzed.utils.isotopeDistributionTable("C4Cl", R=None)
    t._print(out=regtest)
