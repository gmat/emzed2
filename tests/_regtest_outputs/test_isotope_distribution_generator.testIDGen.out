mf       mass       abundance
str      float      float
------   ------     ------
S4C4     175.888283 0.7945
S4C4     176.887902 0.0267
S4C4     176.891533 0.0352
S4C4     177.884079 0.1436
mf       mass       abundance
str      float      float
------   ------     ------
S4C4     175.888283 0.7977
S4C4     176.889972 0.0580
S4C4     177.884079 0.1442
mf       mass       abundance
str      float      float
------   ------     ------
S4C4     179.901703 0.8245
S4C4     180.901090 0.0264
S4C4     181.897499 0.1490
mf       mass      abundance
str      float     float
------   ------    ------
C4       48.000000 0.0625
C4       49.003355 0.2500
C4       50.006710 0.3750
C4       51.010065 0.2500
C4       52.013420 0.0625
mf                    mass      abundance
str                   float     float
------                ------    ------
[12]C4 [35]Cl1        82.968853 0.7257
[12]C3 [13]C1 [35]Cl1 83.972208 0.0314
[12]C4 [37]Cl1        84.965903 0.2322
[12]C3 [13]C1 [37]Cl1 85.969258 0.0100
-                     -         0.0007
